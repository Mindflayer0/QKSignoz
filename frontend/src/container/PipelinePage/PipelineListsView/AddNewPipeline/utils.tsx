import React from 'react';

import { pipelineFields } from '../config';
import DescriptionTextArea from './FormFields/DescriptionTextArea';
import FilterSearch from './FormFields/FilterSearch';
import NameInput from './FormFields/NameInput';
// import ProcessorTags from './FormFields/ProcessorTags';

export const renderPipelineForm = (): // setTagsListData: (tags: Array<string>) => void,
// tagsListData?: Array<string>,
Array<JSX.Element> =>
	pipelineFields.map((fieldName) => {
		if (fieldName.id === '1') {
			return <FilterSearch key={fieldName.id} fieldData={fieldName} />;
		}
		if (fieldName.id === '2') {
			return <NameInput key={fieldName.id} fieldData={fieldName} />;
		}
		/* if (fieldName.id === '3') {
			return (
				<ProcessorTags
					key={fieldName.id}
					fieldData={fieldName}
					setTagsListData={setTagsListData}
					tagsListData={tagsListData || []}
				/>
			);
		} */
		return <DescriptionTextArea key={fieldName.id} fieldData={fieldName} />;
	});
