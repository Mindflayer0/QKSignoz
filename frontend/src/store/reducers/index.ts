import { combineReducers } from 'redux';

import appReducer from './app';
import dashboardReducer from './dashboard';
import globalTimeReducer from './global';
import { LogsReducer } from './logs';
import metricsReducers from './metric';
import { PipelineReducer } from './pipeline';
import { ServiceMapReducer } from './serviceMap';
import traceReducer from './trace';
import { usageDataReducer } from './usage';

const reducers = combineReducers({
	traces: traceReducer,
	usageDate: usageDataReducer,
	globalTime: globalTimeReducer,
	serviceMap: ServiceMapReducer,
	dashboards: dashboardReducer,
	app: appReducer,
	metrics: metricsReducers,
	logs: LogsReducer,
	pipeline: PipelineReducer,
});

export type AppState = ReturnType<typeof reducers>;

export default reducers;
