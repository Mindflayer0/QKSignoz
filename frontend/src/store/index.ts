import { applyMiddleware, compose, createStore } from 'redux';
import type { ThunkMiddleware } from 'redux-thunk';
import thunk from 'redux-thunk';

import reducers, { AppState } from './reducers';

const composeEnhancers =
	(window && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const store = createStore(
	reducers,
	// @TODO Add Type for AppActions also
	composeEnhancers(applyMiddleware(thunk as ThunkMiddleware<AppState>)),
);

export default store;
