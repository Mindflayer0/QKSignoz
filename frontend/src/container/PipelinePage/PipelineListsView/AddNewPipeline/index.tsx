import { Button, Divider, Form, Modal } from 'antd';
import React, { useCallback, useEffect, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { AppState } from 'store/reducers';
import { PipelineData } from 'types/api/pipeline/def';
import AppReducer from 'types/reducer/app';

import { ActionMode, ActionType } from '../../Layouts/Pipeline';
import { ModalButtonWrapper, ModalTitle } from '../styles';
import { getEditedDataSource, getRecordIndex } from '../utils';
import { renderPipelineForm } from './utils';

function AddNewPipeline({
	isActionType,
	setActionType,
	selectedPipelineData,
	setShowSaveButton,
	setCurrPipelineData,
	currPipelineData,
}: AddNewPipelineProps): JSX.Element {
	const [form] = Form.useForm();
	const { t } = useTranslation('pipeline');
	const { user } = useSelector<AppState, AppReducer>((state) => state.app);
	// const [tagsListData, setTagsListData] = useState<Array<string>>();

	const isEdit = isActionType === 'edit-pipeline';
	const isAdd = isActionType === 'add-pipeline';

	useEffect(() => {
		if (isEdit) {
			// setTagsListData(selectedPipelineData?.tags);
			form.setFieldsValue(selectedPipelineData);
		}
		if (isAdd) {
			// setTagsListData([]);
			form.resetFields();
		}
	}, [form, isEdit, isAdd, selectedPipelineData]);

	const onFinish = (values: PipelineData): void => {
		const newPipeLineData: PipelineData = {
			orderId: currPipelineData.length + 1,
			createdAt: new Date().toISOString(),
			createdBy: user?.name || '',
			name: values.name,
			alias: values.name.replace(/\s/g, ''),
			description: values.description,
			filter: values.filter,
			// tags: tagsListData || [],
			config: [],
			enabled: true,
		};

		if (isEdit && selectedPipelineData) {
			const findRecordIndex = getRecordIndex(
				currPipelineData,
				selectedPipelineData,
				'name',
			);
			const updatedPipelineData: PipelineData = {
				...currPipelineData[findRecordIndex],
				name: values.name,
				description: values.description,
				filter: values.filter,
				// tags: tagsListData || [],
			};

			const editedPipelineData = getEditedDataSource(
				currPipelineData,
				selectedPipelineData,
				'name',
				updatedPipelineData,
			);

			setCurrPipelineData(editedPipelineData);
		}
		if (isAdd) {
			// setTagsListData([]);
			setCurrPipelineData((prevState) => [...prevState, newPipeLineData]);
		}
		setActionType(undefined);
	};

	const onCancelModalHandler = (): void => {
		setActionType(undefined);
	};

	const modalTitle = useMemo(
		(): string =>
			isEdit
				? `${t('edit_pipeline')} : ${selectedPipelineData?.name}`
				: t('create_pipeline'),
		[isEdit, selectedPipelineData?.name, t],
	);

	const onOkModalHandler = useCallback(
		() => setShowSaveButton(ActionMode.Editing),
		[setShowSaveButton],
	);

	const isOpen = useMemo(() => isEdit || isAdd, [isAdd, isEdit]);

	return (
		<Modal
			title={<ModalTitle level={4}>{modalTitle}</ModalTitle>}
			centered
			open={isOpen}
			width={800}
			footer={null}
			onCancel={onCancelModalHandler}
		>
			<Divider plain />
			<Form
				name="add-new-pipeline"
				layout="vertical"
				onFinish={onFinish}
				autoComplete="off"
				form={form}
			>
				{/* {renderPipelineForm(setTagsListData, tagsListData)} */}
				{renderPipelineForm()}
				<Divider plain />
				<Form.Item>
					<ModalButtonWrapper>
						<Button
							key="submit"
							type="primary"
							htmlType="submit"
							onClick={onOkModalHandler}
						>
							{isEdit ? t('update') : t('create')}
						</Button>
						<Button key="cancel" onClick={onCancelModalHandler}>
							{t('cancel')}
						</Button>
					</ModalButtonWrapper>
				</Form.Item>
			</Form>
		</Modal>
	);
}

interface AddNewPipelineProps {
	isActionType: string;
	setActionType: (actionType?: ActionType) => void;
	selectedPipelineData: PipelineData | undefined;
	setShowSaveButton: (actionMode: ActionMode) => void;
	setCurrPipelineData: (
		value: React.SetStateAction<Array<PipelineData>>,
	) => void;
	currPipelineData: Array<PipelineData>;
}

export default AddNewPipeline;
