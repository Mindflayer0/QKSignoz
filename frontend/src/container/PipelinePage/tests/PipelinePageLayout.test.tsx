import { render } from '@testing-library/react';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import PipelinePageLayout from '../Layouts';
import { matchMedia } from './AddNewPipeline.test';

beforeAll(() => {
	matchMedia();
});

describe('PipelinePage container test', () => {
	it('should render PipelinePageLayout section', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<PipelinePageLayout />
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
