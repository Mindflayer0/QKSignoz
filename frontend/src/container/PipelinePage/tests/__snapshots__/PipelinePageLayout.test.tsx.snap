// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelinePage container test should render AddNewPipeline section 1`] = `<DocumentFragment />`;

exports[`PipelinePage container test should render PipelinePageLayout section 1`] = `
<DocumentFragment>
  .c0.c0.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-bottom: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.c1.c1 {
  margin-left: 1rem;
}

<div
    class="c0"
  >
    <span
      aria-label="question-circle"
      class="anticon anticon-question-circle"
      role="img"
      style="font-size: 1.3125rem;"
    >
      <svg
        aria-hidden="true"
        data-icon="question-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 708c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm62.9-219.5a48.3 48.3 0 00-30.9 44.8V620c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8v-21.5c0-23.1 6.7-45.9 19.9-64.9 12.9-18.6 30.9-32.8 52.1-40.9 34-13.1 56-41.6 56-72.7 0-44.1-43.1-80-96-80s-96 35.9-96 80v7.6c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V420c0-39.3 17.2-76 48.4-103.3C430.4 290.4 470 276 512 276s81.6 14.5 111.6 40.7C654.8 344 672 380.7 672 420c0 57.8-38.1 109.8-97.1 132.5z"
        />
      </svg>
    </span>
    <button
      class="ant-btn css-dev-only-do-not-override-1i536d8 ant-btn-default c1"
      type="button"
    >
      <span
        aria-label="edit"
        class="anticon anticon-edit"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="edit"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32zm-622.3-84c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9z"
          />
        </svg>
      </span>
      <span>
        enter_edit_mode
      </span>
    </button>
  </div>
  <span
    class="ant-input-group-wrapper ant-input-search css-dev-only-do-not-override-1i536d8"
  >
    <span
      class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-1i536d8"
    >
      <span
        class="ant-input-affix-wrapper css-dev-only-do-not-override-1i536d8"
      >
        <input
          class="ant-input css-dev-only-do-not-override-1i536d8"
          placeholder="search_pipeline_placeholder"
          type="text"
          value=""
        />
        <span
          class="ant-input-suffix"
        >
          <span
            class="ant-input-clear-icon ant-input-clear-icon-hidden"
            role="button"
            tabindex="-1"
          >
            <span
              aria-label="close-circle"
              class="anticon anticon-close-circle"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                />
              </svg>
            </span>
          </span>
        </span>
      </span>
      <span
        class="ant-input-group-addon"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-1i536d8 ant-btn-default ant-btn-icon-only ant-input-search-button"
          type="button"
        >
          <span
            aria-label="search"
            class="anticon anticon-search"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="search"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
  </span>
  .c2 {
  margin: 0.125rem;
  padding: 0.313rem;
  border: none;
  font-style: normal;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.25rem;
}

.c0 {
  margin-top: 3rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.938rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  color: #D89614;
  margin: 0.125rem;
  padding: 0.313rem;
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div>
          Mode: 
          <span>
            Viewing
          </span>
        </div>
        <div>
          Configuration Version 1.0
        </div>
      </div>
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-1i536d8"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-1i536d8"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        class="ant-table-expand-icon-col"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell ant-table-row-expand-icon-cell"
                        />
                        <th
                          class="ant-table-cell"
                          style="text-align: left;"
                        />
                        <th
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          Pipeline Name
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          Filters
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          Tags
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          Last Edited
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          Edited By
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="22a588b8-cccc-4a49-94f1-2caa28271315"
                        draggable="true"
                      >
                        <td
                          class="ant-table-cell ant-table-row-expand-icon-cell"
                        >
                          <span
                            aria-label="right"
                            class="anticon anticon-right"
                            role="img"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                              />
                            </svg>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <span
                            class="ant-avatar ant-avatar-sm ant-avatar-circle css-dev-only-do-not-override-1i536d8"
                            style="background: rgb(22, 104, 220);"
                          >
                            <span
                              class="ant-avatar-string"
                              style="transform: scale(1) translateX(-50%);"
                            >
                              1
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <div
                            class="c2"
                          >
                            Apache common parser
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <span>
                            attributes.source == nginx
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              server
                            </span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              app
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <span>
                            Mon Feb 06 2023
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <span>
                            nitya
                          </span>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="f58ce70c-5f41-48bc-949d-a822e2409257"
                        draggable="true"
                      >
                        <td
                          class="ant-table-cell ant-table-row-expand-icon-cell"
                        >
                          <span
                            aria-label="right"
                            class="anticon anticon-right"
                            role="img"
                            tabindex="-1"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                              />
                            </svg>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <span
                            class="ant-avatar ant-avatar-sm ant-avatar-circle css-dev-only-do-not-override-1i536d8"
                            style="background: rgb(22, 104, 220);"
                          >
                            <span
                              class="ant-avatar-string"
                              style="transform: scale(1) translateX(-50%);"
                            >
                              2
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <div
                            class="c2"
                          >
                            myservice trace parser
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <span>
                            attributes.source == myservice
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              server
                            </span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              app
                            </span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              host
                            </span>
                            <span
                              class="ant-tag ant-tag-magenta css-dev-only-do-not-override-1i536d8"
                            >
                              realse
                            </span>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <span>
                            Sun Feb 05 2023
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: left;"
                        >
                          <span>
                            nityananda
                          </span>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
              <div
                class="ant-table-footer"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
